src/compiler/checker.ts

```diff
@@ -44868,7 +44868,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         const valueSymbol = resolveEntityName(typeName, SymbolFlags.Value, /*ignoreErrors*/ true, /*dontResolveAlias*/ true, location);
         const resolvedSymbol = valueSymbol && valueSymbol.flags & SymbolFlags.Alias ? resolveAlias(valueSymbol) : valueSymbol;
-        isTypeOnly ||= !!valueSymbol?.declarations?.every(isTypeOnlyImportOrExportDeclaration);
+        isTypeOnly ||= !!(valueSymbol && (valueSymbol.declarations?.every(isTypeOnlyImportOrExportDeclaration) || getTypeOnlyAliasDeclaration(valueSymbol, SymbolFlags.Value)));

```

Comment:
I think `valueSymbol.declarations?.every(isTypeOnlyImportOrExportDeclaration)` is actually redundant with `getTypeOnlyAliasDeclaration(valueSymbol, SymbolFlags.Value)`, and only the latter is needed. Since values can’t merge cross-module, if there is a type-only alias declaration resolving to a value, it’s guaranteed that the _only_ value meaning in the symbol is type-only, and therefore there can be no runtime value to reference.

Reply:
@andrewbranch thanks